### PR TITLE
Test coff arch refactor

### PR DESCRIPTION
**Pull request by galenbwill**
COFF loading support complete

### CODE REVIEW

File: arch_x86.cpp [hunk 1]
```diff
@@ -4407,6 +4415,155 @@ class x64ElfRelocationHandler: public RelocationHandler
...
+					break;
+				case PE_IMAGE_REL_I386_SEG12:
+				case PE_IMAGE_REL_I386_SECTION:
+				case PE_IMAGE_REL_I386_SECREL:
+				case PE_IMAGE_REL_I386_TOKEN:
+				case PE_IMAGE_REL_I386_SECREL7:
+				case PE_IMAGE_REL_I386_DIR16:
+				case PE_IMAGE_REL_I386_REL16:
+				default:
+					// By default, PE relocations are correct when not rebased.
```
Comment (plafosse) [inline arch_x86.cpp]: Looks like a copy of a bad comment as we already support relocations idk what we should do about it though

File: arch_x86.cpp [hunk 2]
```diff
@@ -4407,6 +4415,155 @@ class x64ElfRelocationHandler: public RelocationHandler
...
+				}
+				for (auto& reloc : relocTypes)
+					LogWarn("Unsupported COFF relocation: %s", GetRelocationString((COFFx86RelocationType)reloc));
+			}
+		}
+
+		return true;
+	}
+
+	virtual size_t GetOperandForExternalRelocation(const uint8_t* data, uint64_t addr, size_t length,
```
Comment (plafosse) [inline arch_x86.cpp]: Looks like you copied this code, which is fine but it should be unneeded as the base class implements this exactly

File: arch_x86.cpp [hunk 3]
```diff
@@ -354,53 +409,6 @@ static const char* GetRelocationString(Machox64RelocationType relocType)
 	return "Unknown x64 relocation";
 }
 
-// static const char* GetRelocationString(COFFx86RelocationType relocType)
```
Comment (plafosse) [inline arch_x86.cpp:357]: remove commented code

**Review by plafosse — APPROVED** [review]

File: arch_x86.cpp (hunk 3 above)
Comment (galenbwill) [reply arch_x86.cpp:357]: This code actually shouldn't have been commented out, but it was also kind of broken, so it's fixed and uncommented now.

**Pull request merged**